File: BOJ/CPP/단계별로_풀어보기/03_for문/BOJ_2439.cpp
/*
	BOJ 2439 : 별 찍기 - 2

	author : kdgyun
	link https://st-lab.tistory.com
	link https://github.com/kdgyun
	Created by kdgyun
	Copyright © 2021 kdgyun. All rights reserved.
*/

#include <iostream>

using namespace std;

int main(int argc, char const *argv[]) {

	ios_base::sync_with_stdio(false);

	int N;
	cin >> N;

	for (int row = 1; row <= N; row++) {

		// 공백은 N - row개 만큼 출력한다.
		for (int i = 0; i < N - row; i++) {
			cout << ' ';
		}

		// 별은 row개만큼 출력한다.
		for (int i = 0; i < row; i++) {
			cout << '*';
		}

		// 한 행의 출력이 끝나면 개행(줄바꿈)
		cout << '\n';
	}
	return 0;
}

File: BOJ/Java/단계별로_풀어보기/B_03_반복문/BOJ_2439.java
package 단계별로_풀어보기.B_03_반복문;

/**
 * BOJ 2439 : 별 찍기 - 2
 * 
 * @author kdgyun
 * {@link https://st-lab.tistory.com}
 * {@link https://github.com/kdgyun}
 *
 */

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class BOJ_2439 {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(br.readLine());

		
		StringBuilder sb = new StringBuilder();
		for (int i = 1; i <= N; i++) {
			for (int j = 1; j <= N - i; j++) {
				sb.append(' ');
			}
			for (int k = 0; k < i; k++) {
				sb.append('*');
			}
			sb.append('\n');
		}
		System.out.println(sb);
	}

}
